src/components/TodoList/TodoList.tsx

```diff
@@ -1 +1,40 @@
-export const TodoList = () => {};
+import React from 'react';
+import { TodoInfo } from '../TodoInfo';
+
+interface Todo {
+  id: number;
+  title: string;
+  userId: number;
+  completed: boolean;
+}
+
+interface User {
+  id: number;
+  name: string;
+  email: string;
+}
+
+interface TodoListProps {
+  todos: Todo[];
+  users: User[];
+}
+
+export const TodoList: React.FC<TodoListProps> = ({ todos, users }) => (
+  <section className="TodoList">
+    {todos.map(todo => {
+      const matchedUser = users.find(u => u.id === todo.userId);
+
+      return (
+        <TodoInfo
+          key={todo.id}
+          id={todo.id}
+          title={todo.title}
+          completed={todo.completed}
+          userId={todo.userId}
+          userEmail={matchedUser?.email || ''}
+          userName={matchedUser?.name || 'Unknown User'}
```

Comment:
The use of optional chaining (`matchedUser?.email || ''` and `matchedUser?.name || 'Unknown User'`) is a good way to handle cases where a user might not be found. However, ensure that this behavior is acceptable for your application's requirements, as it might mask data integrity issues if todos are expected to always have a corresponding user.